js/toolbar/z-plane-list.js

```diff
@@ -154,29 +176,85 @@ function ZPlaneList(parent) {
         var loader = _self._zPlaneContainer.querySelector('#z-plane-loader');
         if (loader != null) {
             var carousel = _self._zPlaneContainer.querySelector('.z-plane-carousel');
+            var zIndexInput = _self._zPlaneContainer.querySelector('#main-image-z-index');
+            var zIndexSubmit = _self._zPlaneContainer.querySelector('#z-index-jump-button');
             if (display == true) {
                 carousel.style.pointerEvents = 'none';
                 loader.style.display = 'flex';
-
+                zIndexInput.disabled = true;
+                zIndexSubmit.disabled = true;
             } else {
                 carousel.style.pointerEvents = 'initial';
                 loader.style.display = 'none';
+                zIndexInput.disabled = false;
+                zIndexSubmit.disabled = false;
             }
         }
     };
 
+    /**
+     * change the currentZPlaneRequest variable and fetch the data
+     * @param {string} requestType - must be present in validFetchRequests variable
+     * @param {object} requestData - the content of the fetch request. Make sure the requestData is valid, no such check is done in this function
+     */
+    this._updateCurrentZPlaneRequestAndFetchData = function (requestType, requestData) {
+        _self.currentZPlaneRequest = {
+            type: requestType,
+            data: requestData,
+        };
+
+        if (_self.validFetchRequests.includes(requestType)) {
+            _self.parent.dispatchEvent(requestType, requestData);
+            _self._showSpinner(true);
+        }
+        // console.log('_self.currentZPlaneRequest', _self.currentZPlaneRequest);
+    }
+
     /**
      * calculate the page size if needed and ask chaise to fetch the new list
      */
     this._fetchList = function (data) {
-        _self._showSpinner(true);
 
-        _self.parent.dispatchEvent('fetchZPlaneList', {
+        var requestData = {
             pageSize: _self.pageSize,
             before: data.before,
             after: data.after,
             requestID: ++_self._currentRequestID
-        });
+        };
+
+        _self._updateCurrentZPlaneRequestAndFetchData('fetchZPlaneList', requestData);
+    };
+
+    /**
+     * calculate the page size if needed and ask chaise to fetch the new list by zIndex
+     * @param {string} zIndex
+     */
+    this._fetchListByZIndex = function (zIndex) {
+        // removing leading zeros
+        zIndex = zIndex.replace(/^0+/, '');
+
+        if (zIndex == 'default') {
+            // This is case during the init function
+
+            var requestData = {
+                pageSize: _self.pageSize,
+                zIndex: _self.mainImageZIndex,
+                requestID: ++_self._currentRequestID
+            };
+
+            _self._updateCurrentZPlaneRequestAndFetchData('fetchZPlaneListByZIndex', requestData);
+        } else if (parseInt(zIndex) == _self.mainImageZIndex) {
```

Comment:
instead of an empty `else if` and `else` , just do `else if` of opposite of the condition that you have.

Reply:
Done